hooks/useProposalSignature.ts

```diff
@@ -1,26 +1,27 @@
+import { PublicKey } from '@solana/web3.js'
 import { useState, useEffect } from 'react'
 import useWalletStore from 'stores/useWalletStore'
 import useProposal from './useProposal'
 
 const useProposalSignature = () => {
-  const { proposal } = useProposal()
+  const { proposal, instructions } = useProposal()
   const connection = useWalletStore((s) => s.connection)
   const [proposalSignature, setProposalSignature] = useState('')
-
+  const [itx] = Object.keys(instructions)
```

Comment:
so the point is to take the first instruction? Or any instruction works? Or is there only one instruction in the object?

`Object.keys()` does not guarantee the order

Reply:
here any itx works. There can be one or more itxs in the object depending on how many itxs there is in the proposal, but they all have in common two txs: The creation and the execution

Reply:
actually the thing is:

- if there are 3 instructions that has been executed individually in separated transactions clicking on the "execute" button. Then you get the signature about a random instructions of the 3.

- if there are 3 instructions that has been executed with the "execute all" button, then it's fine

- if there are 1 instruction, then it's fine